tests/test_url_build.py

```diff
@@ -239,6 +239,11 @@ def test_build_with_authority_with_path_without_leading_slash():
         URL.build(scheme="http", host="example.com", path="path_without_leading_slash")
 
 
+def test_build_with_none_host():
+    with pytest.raises(TypeError):
```

Comment:
It is always important to use the `match` argument with `pytest.raises()`.

Reply:
Added a `match` now but none of the other test cases uses it (just FYI).

Reply:
That's unfortunate. And a chance to contribute more refactoring PRs for someone.